kernel/src/syscall/xattr.rs

```diff
@@ -0,0 +1,421 @@
+
+use alloc::borrow::Cow;
+
+use super::SyscallReturn;
+use crate::{
+    fs::{
+        file_handle::FileLike,
+        file_table::{get_file_fast, FileDesc},
+        fs_resolver::{FsPath, AT_FDCWD},
+        path::Dentry,
+        utils::{
+            XattrName, XattrNamespace, XattrSetFlags, XATTR_LIST_MAX_LEN, XATTR_NAME_MAX_LEN,
+            XATTR_VALUE_MAX_LEN,
+        },
+    },
+    prelude::*,
+    process::credentials::capabilities::CapSet,
+    syscall::constants::MAX_FILENAME_LEN,
+};
+
+pub fn sys_setxattr(
+    path_ptr: Vaddr,
+    name_ptr: Vaddr,
+    value_ptr: Vaddr,
+    value_len: usize,
+    flags: i32,
+    ctx: &Context,
+) -> Result<SyscallReturn> {
+    let user_space = ctx.user_space();
+    let path = user_space.read_cstring(path_ptr, MAX_FILENAME_LEN)?;
+
+    setxattr(
+        XattrFileCtx::Path(path),
+        name_ptr,
+        value_ptr,
+        value_len,
+        flags,
+        &user_space,
+        ctx,
+    )?;
+
+    Ok(SyscallReturn::Return(0))
+}
+
+pub fn sys_lsetxattr(
+    path_ptr: Vaddr,
+    name_ptr: Vaddr,
+    value_ptr: Vaddr,
+    value_len: usize,
+    flags: i32,
+    ctx: &Context,
+) -> Result<SyscallReturn> {
+    let user_space = ctx.user_space();
+    let path = user_space.read_cstring(path_ptr, MAX_FILENAME_LEN)?;
+
+    setxattr(
+        XattrFileCtx::PathNoFollow(path),
+        name_ptr,
+        value_ptr,
+        value_len,
+        flags,
+        &user_space,
+        ctx,
+    )?;
+
+    Ok(SyscallReturn::Return(0))
+}
+
+pub fn sys_fsetxattr(
+    fd: FileDesc,
+    name_ptr: Vaddr,
+    value_ptr: Vaddr,
+    value_len: usize,
+    flags: i32,
+    ctx: &Context,
+) -> Result<SyscallReturn> {
+    let mut file_table = ctx.thread_local.file_table().borrow_mut();
+    let file = get_file_fast!(&mut file_table, fd);
+
+    let user_space = ctx.user_space();
+    setxattr(
+        XattrFileCtx::FileHandle(file),
+        name_ptr,
+        value_ptr,
+        value_len,
+        flags,
+        &user_space,
+        ctx,
+    )?;
+
+    Ok(SyscallReturn::Return(0))
+}
+
+fn setxattr(
+    file_ctx: XattrFileCtx,
+    name_ptr: Vaddr,
+    value_ptr: Vaddr,
+    value_len: usize,
+    flags: i32,
+    user_space: &CurrentUserSpace,
+    ctx: &Context,
+) -> Result<()> {
+    let flags = XattrSetFlags::from_bits(flags as _)
+        .ok_or(Error::with_message(Errno::EINVAL, "invalid xattr flags"))?;
+
+    let name_cstr = read_xattr_name_cstr_from_user(name_ptr, user_space)?;
+    let name_str = name_cstr.to_string_lossy();
+    let xattr_name = parse_xattr_name(name_str.as_ref())?;
+    check_namespace(xattr_name.namespace(), ctx)?;
+
+    if value_len > XATTR_VALUE_MAX_LEN {
+        return_errno_with_message!(Errno::E2BIG, "xattr value too long");
+    }
+    let mut value_reader = user_space.reader(value_ptr, value_len)?;
+
+    let dentry = lookup_target_dentry(&file_ctx, ctx)?;
+    dentry.set_xattr(xattr_name, &mut value_reader, flags)
+}
+
+pub fn sys_getxattr(
```

Comment:
Could you follow the established convention to give each system call a separate file?

Some examples are listed below:
* `sched_get_priority_max.rs` and `sched_get_priority_min.rs`
* `getuid.rs` and `setuid.rs`
* `read.rs`, `preadv.rs`, and `preadv64.rs`

In addition, the addition of new system calls should be put in a single commit.